code/game/objects/machinery/overwatch.dm

```diff
@@ -658,6 +663,18 @@ GLOBAL_LIST_EMPTY(active_cas_targets)
 	visible_message(span_boldnotice("[transfer_marine] has been transfered from squad '[old_squad]' to squad '[new_squad]'. Logging to enlistment file."))
 	to_chat(transfer_marine, "[icon2html(src, transfer_marine)] <font size='3' color='blue'><B>\[Overwatch\]:</b> You've been transfered to [new_squad]!</font>")
 
+///Big light :)
+/obj/machinery/computer/camera_advanced/overwatch/proc/attempt_spotlight(datum/source, atom/A, params)
+	SIGNAL_HANDLER
+	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_ORBITAL_SPOTLIGHT))
+		to_chat(source, span_notice("The Orbital spotlight is still recharging."))
+		return
+	var/turf/target = get_turf(A)
```

Comment:
Also, I see no check for roofing in these proc calls.

Unless command is teleporting a sun into a building, there should be a check for the orbital spotlight for opaque roofs (metal, underground, deep underground).

Reply:
I added a check stolen from the mortar